framework/test_framework.py

```diff
@@ -100,10 +101,19 @@ def deploy_test(platform):
     if platform in ["qemu-aarch64-virt", "qemu-riscv64-virt"]:
         arch = platform.split("-")[1]
         run_cmd = "../platform/qemu/run.sh " + arch
+
+        ports_init = connection.scan_pts_ports()
         process = run_command_in_terminal(run_cmd)
-        # Connection to platform will run here
-        # which will stall the platform untill the results
-        # are sent from the platform to the framework
+
+        ports_aux = connection.scan_pts_ports()

```

Comment:
When executing the test framework on Qemu, it's essential to utilize pseudo-terminals (`/dev/pts/`) instead of physical USB ports (`/dev/tty`). This choice ensures the possibility of establishing an SSH connection to the emulator if needed.

In preparation for launching Qemu, the following code snippet monitors the ports in use (`ports_aux`) and leverages the `scan_pts_ports()` function to identify the pseudo-terminals opened by Qemu. This process updates the `ports_end `variable. Refactoring the variable names for clarity might enhance readability; for instance, do you think that renaming them to `pts_ports_before `and `pts_ports_after `could improve understanding and maintainability?

Reply:
Fully agree, thanks for your feedback! Addressed in https://github.com/bao-project/bao-tests/pull/13/commits/d281815cc78998162b5537e118a9d1dbe50a8ec0.